File: src/main/java/com/methodpark/cmpe/code/lsp/Duck.java
package com.methodpark.cmpe.code.lsp;

public abstract class Duck {
    public void quack() {
        System.out.println("quaak");
    }
    
    public void swim() {
        System.out.println("splash splash");
    }
    
    public abstract void display();
    
    public void fly() {
        System.out.println("flap flap flap");
    }
}

File: src/main/csharp/Lsp/Lsp/Duck.cs
using System;

namespace Lsp
{
    public abstract class Duck
    {
        public virtual void Quack()
        {
            Console.WriteLine("quaak");
        }

        public virtual void Swim()
        {
            Console.WriteLine("splash splash");
        }

        public abstract void Display();

        public virtual void Fly()
        {
            Console.WriteLine("flap flap flap");
        }
    }
}